xarray/core/datatree.py

```diff
@@ -1238,65 +1238,67 @@ def isomorphic(
         except (TypeError, TreeIsomorphismError):
             return False
 
-    def equals(self, other: DataTree, from_root: bool = True) -> bool:
+    def _matching(
+        self,
+        other: DataTree,
+        nodes_match: Callable[[DataTree, DataTree], bool],
+    ) -> bool:
+        if not self.isomorphic(other, from_root=True, strict_names=True):
+            return False
+
+        return all(
+            nodes_match(node, other_node)
+            for node, other_node in zip(self.descendants, other.descendants)
+        )
+
+    def equals(self, other: DataTree) -> bool:
         """
         Two DataTrees are equal if they have isomorphic node structures, with matching node names,
         and if they have matching variables and coordinates, all of which are equal.
 
-        By default this method will check the whole tree above the given node.
-
         Parameters
         ----------
         other : DataTree
             The other tree object to compare to.
-        from_root : bool, optional, default is True
-            Whether or not to first traverse to the root of the two trees before checking for isomorphism.
-            If neither tree has a parent then this has no effect.
 
         See Also
         --------
         Dataset.equals
         DataTree.isomorphic
         DataTree.identical
         """
-        if not self.isomorphic(other, from_root=from_root, strict_names=True):
+        to_ds = lambda x: x._to_dataset_view(inherited=True, rebuild_dims=False)
```

Comment:
Wouldn't it be simpler (and clearer) to use `inherited=False`? The nodes are checked in depth-first order anyway, so if there is a discrepancy between inherited coordinates it will be discovered at the top of the subtree before checking the children.